                                                                                       MULTITHREADING IN JAVA
AIM: 
      To write a Java program to demonstrate multithreading using Thread, yield(), and sleep() methods.
ALGORITHM :
Start the program.
Create three threads: A, B, and C.
In thread A, display numbers from 1 to 5 and use yield().
In thread B, display numbers from 1 to 3 and then terminate the thread.
In thread C, display numbers from 1 to 5 and use sleep() for 1.5 seconds after displaying 1.
Start all three threads using the start() method.
Display the message from the main thread.
Stop the program.

PROGRAM:
    
class A extends Thread {
    public void run() {
        for (int i = 1; i <= 5; i++) {
            if (i == 1) {
                Thread.yield();
            }

            System.out.println("from thread A i=" + i);
        }

        System.out.println("exit from A");
    }
}

class B extends Thread {
    public void run() {
        for (int j = 1; j <= 5; j++) {
            System.out.println("from thread B j=" + j);

            if (j == 3) {
                System.out.println("exit from B");
                break;
            }
        }
    }
}

class C extends Thread {
    public void run() {
        for (int k = 1; k <= 5; k++) {
            System.out.println("thread C = " + k);

            if (k == 1) {
                try {
                    Thread.sleep(1500);
                } catch (InterruptedException e) {
                    System.out.println("Thread C interrupted");
                }
            }
        }
    }
}

public class Threadtest {

    public static void main(String[] args) {
        A a = new A();
        B b = new B();
        C c = new C();

        System.out.println("Start thread A");

        a.start();
        b.start();
        c.start();

        System.out.println("exit from main thread");
    }
}

 OUTPUT:
 Start thread A
exit from main thread
from thread A i=1
from thread A i=2
from thread A i=3
from thread A i=4
from thread A i=5
exit from A
from thread B j=1
from thread B j=2
from thread B j=3
exit from B
thread C = 1
thread C = 2
thread C = 3
thread C = 4
thread C = 5

RESULT:
       Thus, the Java program to demonstrate multithreading using yield() and sleep() methods was successfully executed.
